### PR TITLE
detangle kazoo_media and kazoo_call

### PR DESCRIPTION
**Pull request by jamesaimonetti**
move kz_media_recording into kazoo_call as kzc_recording

update usage of kazoo_media modules to not use Call; callers must pass
in account id or language directly and not rely on kapps_call function
calls.

resolve a lot of dialyzer complaints

### CODE REVIEW

**Comment by lazedo** [comment]
@jamesaimonetti mkbusy fails due to `language` support

```
before
[info] [<0.12351.0>] 172.20.0.6 - - GET /system_media/en-mb%2fvm-enter_pass 200
[info] [<0.12341.0>] 172.20.0.6 - - GET /system_media/en-mb%2fvm-enter_pass?stream_type=single&doc_type=media&rev=2-de245a47d752ec7e67c688dc44325068 200
```

```
after
[info] [<0.12422.0>] 172.19.0.6 - - GET /system_media/en-us%2fvm-enter_pass 200
[info] [<0.12422.0>] 172.19.0.6 - - GET /system_media/en-us%2fvm-enter_pass?stream_type=single&doc_type=media&rev=2-f07d14c8f5c1540ea2f479233a13b246 200
```

File: applications/callflow/src/module/cf_call_forward.erl [hunk 1]
```diff
@@ -87,8 +87,8 @@ cf_menu(#callfwd{keys=#keys{menu_toggle_cf=Toggle
                 }=CF, CaptureGroup, Call) ->
     lager:info("playing call forwarding menu"),
     Prompt = case Enabled of
-                 'true' -> kz_media_util:get_prompt(<<"cf-enabled_menu">>, Call);
-                 'false' -> kz_media_util:get_prompt(<<"cf-disabled_menu">>, Call)
+                 'true' -> kz_media_util:get_prompt(<<"cf-enabled_menu">>, kapps_call:account_id(Call));
+                 'false' -> kz_media_util:get_prompt(<<"cf-disabled_menu">>, kapps_call:account_id(Call))
```
Comment (lazedo) [inline applications/callflow/src/module/cf_call_forward.erl]: @jamesaimonetti i think this should pass the `Language` of the call and not the accountid

File: applications/callflow/src/module/cf_call_forward.erl [hunk 2]
```diff
@@ -187,7 +187,7 @@ cf_deactivate(CF, Call) ->
 -spec cf_update_number(callfwd(), api_binary(), kapps_call:call()) -> callfwd().
 cf_update_number(#callfwd{interdigit_timeout=Interdigit}=CF, CaptureGroup, Call)
   when is_atom(CaptureGroup); CaptureGroup =:= <<>> ->
-    EnterNumber = kz_media_util:get_prompt(<<"cf-enter_number">>, Call),
+    EnterNumber = kz_media_util:get_prompt(<<"cf-enter_number">>, kapps_call:account_id(Call)),
```
Comment (lazedo) [inline applications/callflow/src/module/cf_call_forward.erl]: same here

File: applications/callflow/src/module/cf_menu.erl [hunk 3]
```diff
@@ -361,7 +360,7 @@ play_exit_prompt(#cf_menu_data{exit_media=Id}, Call) ->
 %%--------------------------------------------------------------------
 -spec get_prompt(menu(), kapps_call:call()) -> ne_binary().
 get_prompt(#cf_menu_data{greeting_id='undefined'}, Call) ->
-    kz_media_util:get_prompt(<<"menu-no_prompt">>, Call);
+    kz_media_util:get_prompt(<<"menu-no_prompt">>, kapps_call:account_id(Call));
```
Comment (lazedo) [inline applications/callflow/src/module/cf_menu.erl]: another

File: core/kazoo_endpoint/src/kz_endpoint.erl [hunk 4]
```diff
@@ -62,7 +62,9 @@
 -define(DEFAULT_MOBILE_AMQP_CONNECTIONS,
         kz_json:from_list([{<<"default">>, ?DEFAULT_MOBILE_AMQP_CONNECTION}])).
 
--define(CONFIRM_FILE(Call), kz_media_util:get_prompt(<<"ivr-group_confirm">>, Call)).
+-define(CONFIRM_FILE(Call)
+       ,kz_media_util:get_prompt(<<"ivr-group_confirm">>, kapps_call:account_id(Call))
```
Comment (lazedo) [inline core/kazoo_endpoint/src/kz_endpoint.erl]: same here. every former `kz_media_util:get_prompt/2` passing a `Call` object should now call `kz_media_util:get_prompt/3` with `language` and `account_id` of the `Call`. maybe create a `kapps_call:get_prompt/2` (prompt, Call) and havew it call  `kz_media_util:get_prompt/3` ?

Comment (jamesaimonetti) [inline core/kazoo_endpoint/src/kz_endpoint.erl]: Good idea, will update soon!

**Pull request merged**